### PR TITLE
scorecard model

### PR DESCRIPTION
**Pull request by Yancey0623**

### CODE REVIEW

File: sqlflow_models/score_card.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#!/bin/env python
```
Comment (typhoonzero) [inline sqlflow_models/score_card.py:1]: Add model test code?

File: sqlflow_models/score_card.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    def _woe_encoder(self, x, y):
+        x_train_dict = {}
+        for col in x.columns:
+            dfx, cut, woe, iv = self._pf_bin(y, x[col])
+            self._bins[col] = (dfx, cut, woe, iv)
+            # replacing by the WOE encode
+            x_train_dict[col] = self._replace_woe(x[col], cut, woe)
+        return pd.DataFrame.from_dict(x_train_dict)
+
+    def sqlflow_train_loop(self, dataset, epochs=1, verbose=0):
```
Comment (typhoonzero) [inline sqlflow_models/score_card.py:85]: Also define `sqlflow_predict_one` to do prediction later?

**Review by typhoonzero — APPROVED** [review]

File: sqlflow_models/score_card.py (hunk 2 above)
Comment (Yancey0623) [reply sqlflow_models/score_card.py:85]: Yes will add it if needed.The training process would print the score card as follows, 

``` text
 THE SCORE CARD:
 age (20.999, 38.0] 3.0
 age (38.0, 47.0] 5.0
 age (47.0, 56.0] -1.0
 age (56.0, 65.0] -3.0
 age (65.0, 97.0] -9.0
 debt_ratio (-0.001, 0.131] -11.0
 debt_ratio (0.131, 0.286] -0.0
 debt_ratio (0.286, 0.474] -7.0
 debt_ratio (0.474, 3.248] 11.0
 debt_ratio (3.248, 15466.0] -1.0
 monthly_income (-0.001, 3325.8] 3.0
 monthly_income (3325.8, 5212.6] 5.0
 monthly_income (5212.6, 5400.0] -3.0
 monthly_income (5400.0, 8169.4] -3.0
 monthly_income (8169.4, 208333.0] -4.0
 number_of_dependents (-0.001, 2.0] 0.0
 number_of_dependents (2.0, 8.0] -3.0
 number_of_open_credit_lines_and_loans (-0.001, 4.0] 4.0
 number_of_open_credit_lines_and_loans (4.0, 6.0] -1.0
 number_of_open_credit_lines_and_loans (6.0, 9.0] -4.0
 number_of_open_credit_lines_and_loans (9.0, 12.0] -4.0
 number_of_open_credit_lines_and_loans (12.0, 46.0] 4.0
 number_of_time30_59days_past_due_not_worse (-0.001, 98.0] 0.0
 number_of_time60_89_days_past_due_not_worse (-0.001, 98.0] 0.0
 number_of_times_90_days_late (-0.001, 98.0] 0.0
 number_real_estate_loans_or_lines (-0.001, 1.0] -2.0
 number_real_estate_loans_or_lines (1.0, 2.0] -1.0
 number_real_estate_loans_or_lines (2.0, 9.0] 18.0
 revolving_utilization_of_unsecured_lines (-0.001, 0.0188] -35.0
 revolving_utilization_of_unsecured_lines (0.0188, 0.0866] -44.0
 revolving_utilization_of_unsecured_lines (0.0866, 0.273] -39.0
 revolving_utilization_of_unsecured_lines (0.273, 0.686] -10.0
 revolving_utilization_of_unsecured_lines (0.686, 2340.0] 43.0
```

File: sqlflow_models/score_card.py (hunk 1 above)
Comment (Yancey0623) [reply sqlflow_models/score_card.py:1]: done

**Pull request merged**